salt/states/file.py

```diff
@@ -3516,69 +3520,65 @@ def append(name,
     with salt.utils.fopen(name, 'rb') as fp_:
         slines = fp_.readlines()
 
-    count = 0
-    test_lines = []
-
+    append_lines = []
     try:
         for chunk in text:
             if ignore_whitespace:
                 if __salt__['file.search'](
-                    name,
-                    salt.utils.build_whitespace_split_regex(chunk),
-                    multiline=True):
+                        name,
+                        salt.utils.build_whitespace_split_regex(chunk),
+                        multiline=True):
                     continue
             elif __salt__['file.search'](
                     name,
                     chunk,
                     multiline=True):
                 continue
 
-            lines = chunk.splitlines()
+            for line_item in chunk.splitlines():
+                append_lines.append('{0}'.format(line_item))
 
-            for line in lines:
-                if __opts__['test']:
-                    ret['comment'] = 'File {0} is set to be updated'.format(name)
-                    ret['result'] = None
-                    test_lines.append('{0}\n'.format(line))
-                else:
-                    __salt__['file.append'](name, line)
-                count += 1
     except TypeError:
         return _error(ret, 'No text found to append. Nothing appended')
 
     if __opts__['test']:
-        nlines = slines + test_lines
+        ret['comment'] = 'File {0} is set to be updated'.format(name)
         ret['result'] = None
+        nlines = list(slines)
+        nlines.extend(append_lines)
         if slines != nlines:
             if not salt.utils.istextfile(name):
                 ret['changes']['diff'] = 'Replace binary file'
             else:
                 # Changes happened, add them
                 ret['changes']['diff'] = (
-                    ''.join(difflib.unified_diff(slines, nlines))
+                    '\n'.join(difflib.unified_diff(slines, nlines))
                 )
         else:
             ret['comment'] = 'File {0} is in correct state'.format(name)
             ret['result'] = True
         return ret
 
+    if len(append_lines):
```

Comment:
right, but an empty list evaluates to False, and a list with stuff evaluates to True, making the len unnecessary:
https://docs.python.org/2/library/stdtypes.html#truth-value-testing